examples/vision/swin_transformers.py

```diff
@@ -0,0 +1,557 @@
+            self.relative_position_index, shape=(-1,)
+        )
+        relative_position_bias = tf.gather(
+            self.relative_position_bias_table, relative_position_index_flat
+        )
+        relative_position_bias = tf.reshape(
+            relative_position_bias, shape=(num_window_elements, num_window_elements, -1)
+        )
+        relative_position_bias = tf.transpose(relative_position_bias, perm=(2, 0, 1))
+        attn = attn + tf.expand_dims(relative_position_bias, axis=0)
+
+        if mask is not None:
+            nW = mask.get_shape()[0]
+            mask_float = tf.cast(
+                tf.expand_dims(tf.expand_dims(mask, axis=1), axis=0), tf.float32
+            )
+            attn = (
+                tf.reshape(attn, shape=(-1, nW, self.num_heads, size, size))
+                + mask_float
+            )
+            attn = tf.reshape(attn, shape=(-1, self.num_heads, size, size))
+            attn = keras.activations.softmax(attn, axis=-1)
+        else:
+            attn = keras.activations.softmax(attn, axis=-1)
+        attn = self.dropout(attn)
+
+        x_qkv = attn @ v
+        x_qkv = tf.transpose(x_qkv, perm=(0, 2, 1, 3))
+        x_qkv = tf.reshape(x_qkv, shape=(-1, size, channels))
+        x_qkv = self.proj(x_qkv)
+        x_qkv = self.dropout(x_qkv)
+        return x_qkv
+
+
+"""
+## The final Swin Transformer model
+
+We will now put together a Swin Transformer by replacing the standard multi-head 
+self attention (MSA) in a Transformer with shifted windows. As suggested in the 
+original paper we will create a model comprising of a shifted window based MSA
+layer, followed by a 2-layer MLP with GELU nonlinearity in between, applying 
+`LayerNormalization` before each MSA layer and each MLP, and a residual 
+connection after each of these layers.
+"""
+
+
+class SwinTransformer(layers.Layer):
+    def __init__(
+        self,
+        dim,
+        num_patch,
+        num_heads,
+        window_size=7,
+        shift_size=0,
+        num_mlp=1024,
+        qkv_bias=True,
+        dropout_rate=0.0,
+        **kwargs,
+    ):
+        super(SwinTransformer, self).__init__(**kwargs)
+
+        self.dim = dim  # number of input dimensions
+        self.num_patch = num_patch  # number of embedded patches
+        self.num_heads = num_heads  # number of attention heads
+        self.window_size = window_size  # size of window
+        self.shift_size = shift_size  # size of window shift
+        self.num_mlp = num_mlp  # number of MLP nodes
+
+        self.norm1 = layers.LayerNormalization(epsilon=1e-5)
+        self.attn = WindowAttention(
+            dim,
+            window_size=(self.window_size, self.window_size),
+            num_heads=num_heads,
+            qkv_bias=qkv_bias,
+            dropout_rate=dropout_rate,
+        )
+        self.drop_path = DropPath(dropout_rate)
+        self.norm2 = layers.LayerNormalization(epsilon=1e-5)
+        self.mlp = Mlp([num_mlp, dim], drop=dropout_rate)
```

Comment:
that is to say, here

Reply:
Got it, put the MLP definition inline in this code block